engine/src/main/java/org/hibernate/search/backend/DeletionQuery.java

```diff
@@ -15,10 +19,23 @@
 public interface DeletionQuery {
 
 	/**
-	 * used to identify the type of query faster (no need for instanceof checks) in the Delegate
-	 *
+	 * used to identify the type of query faster (no need for instanceof checks)
 	 * @return the unique query key
 	 */
 	int getQueryKey();
 
+	/**
+	 * converts this DeletionQuery to a Lucene Query
+	 *
+	 * @param analyzerForEntity the analyzer to be used for this query
+	 */
+	Query toLuceneQuery(ScopedAnalyzer analyzerForEntity);
```

Comment:
@Sanne: I'm wondering whether this interface (and its impl) are rightly located in an API package. Isn't it more SPI? Also how strict are we about breaking changes here. I don't think adding these new methods is a big deal, as it all seems quite experimental to me anyways (should we tag it as such btw.), but WDYT?